flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/shuffle/TieredResultPartitionTest.java

```diff
@@ -192,15 +188,34 @@ void testMetricsUpdateForBroadcastOnlyResultPartition() throws Exception {
         try (TieredResultPartition partition =
                 createTieredStoreResultPartition(2, bufferPool, true)) {
             partition.broadcastRecord(ByteBuffer.allocate(bufferSize));
-            IOMetrics ioMetrics = taskIOMetricGroup.createSnapshot();
-            assertThat(ioMetrics.getResultPartitionBytes()).hasSize(1);
-            ResultPartitionBytes partitionBytes =
-                    ioMetrics.getResultPartitionBytes().values().iterator().next();
-            assertThat(partitionBytes.getSubpartitionBytes())
-                    .containsExactly(bufferSize, bufferSize);
+            verifySubpartitionBytes(bufferSize, bufferSize);
         }
     }
 
+    @Test
+    void testRequestBuffersAfterPoolSizeDecreased() throws IOException {
```

Comment:
Good catch. The test would fail before this PR at first, but I forgot to keep the behavior after some changes which made it not expected. I've rollbacked it, please take a look.